### PR TITLE
Accepting `custom fs` as an argument for pluggable `fs` feature.

### PR DESCRIPTION
**Pull request by SparshithNR**

### CODE REVIEW

File: lib/source-map.js [hunk 1]
```diff
@@ -311,13 +315,11 @@ class SourceMap {
     } else {
       // Look for original sources relative to our input source filename.
       return srcMap.sources.map(function(source){
-        let fullPath;
         if (path.isAbsolute(source)) {
-          fullPath = source;
+          return ensurePosix(fs.readFileSync(source, 'utf-8'));
```
Comment (stefanpenner) [inline lib/source-map.js]: @SparshithNR `fs`?

File: lib/source-map.js [hunk 2]
```diff
@@ -74,7 +68,12 @@ class SourceMap {
 
 
   addFile(filename) {
-    let source = ensurePosix(fs.readFileSync(this._resolveFile(filename), 'utf-8'));
+    let source = '';
+    if (path.isAbsolute(filename)) {
+      source = ensurePosix(fs.readFileSync(filename));
```
Comment (stefanpenner) [inline lib/source-map.js]: Let's consider moving this logic to the input abstraction, then this library merely needs to support a pluggable fs.

File: lib/source-map.js [hunk 3]
```diff
@@ -22,6 +22,7 @@ class SourceMap {
       throw new Error("must specify the mapURL when setting a custom mapFile");
     }
     this.baseDir = opts.baseDir;
+    this.input = opts.input;
```
Comment (stefanpenner) [inline lib/source-map.js]: `this.fs = opts.fs`

File: .travis.yml [hunk 4]
```diff
@@ -1,6 +1,4 @@
 language: node_js
 node_js:
```
Comment (stefanpenner) [inline .travis.yml]: let's do this as a separate PR.

File: lib/source-map.js (hunk 3 above)
Comment (SparshithNR) [reply lib/source-map.js]: done. I have changed it to custom fs.

File: .travis.yml (hunk 4 above)
Comment (SparshithNR) [reply .travis.yml]: done

**Comment by stefanpenner** [comment]
released as 2.1.0
